### DISCUSSION TITLE
请教enrle后RGM色差过大的问题

### DISCUSSION THREAD

**Discussion by kaoskey** [palxex/palresearch#10] [Q&A]
尝试替换土星版立绘，但是enrle重新打包后和土星原版色差太大，达不到土星版的色彩。enrle采用以下命令

`./enrle.py icon.png -o ABC330.rle -p PAT.MKF --quantize`

如果加了--dither，会导致彩色噪点过多。
![palss -1](https://user-images.githubusercontent.com/87409689/125577886-2d62fe83-6031-4036-9909-51b1d54acc7d.png)

求教各位大佬是否有解。

**Comment by palxex** [comment]
暂时是无解的，问题出在土星版的调色板与DOS版相差过大，而color dithering对这个分辨率效果比较一般。现在只能把导入后的图重新导出，然后在usenti里手动修改颜色，最后重新导入，整个过程中避免再次发生color quantize以保证不会再次发生颜色变动。未来可以看看SDLPal会不会直接支持全彩图，这个是一劳永逸的做法，但时间很难预期。
另外发现discussions貌似并不会自动提醒，以后急需获得回答的话可以尝试@我一下。

**Reply by kaoskey** [reply]
@palxex 感谢大佬回答。现在看来只能用着色器稍作美化能减轻这种差异。另外想请教，rle前的图片用什么格式能尽量减轻杂色，png-8，256位，png-16 srgb还是bmp? 还是其实都没什么差别。再次感谢！

**Reply by palxex** [reply]
源格式的话，尽可能颜色丰富点可能转出来还好一些，毕竟原调色板覆盖面还不错。两个调色板之间互相转换图片几乎是最难的，非常看两者本身的相性，因为都是「被优化限定过的颜色」，原始图（如果有）其实反而未必那么难转。
另一方面，从其他测试者的反馈来看，源自其他游戏的高分辨率图片压缩到仙剑的分辨率（320x200）多半会显得糊，因为信息量下降太大，难以避免。这个也很难在原始框架下得到解决，shader锐化的话基本肯定会改变风格。
现在来讲最靠谱的方案其实反而是用能指定调色板的app（比如usenti/fireworks）直接在该分辨率下尝试自己绘制，原始调色板可以用depat.py从资源中提取。

**Reply by kaoskey** [reply]
非常感谢您的回答

**Reply by kaoskey** [reply]
@palxex 鲁痴大佬再次打扰。直接用Usenti改图效果确实好太多了，但是我depat.py使用出了点问题。demkf.py pat.mkf以后出来了9个pat文件，但是每个pat尝试下面的命令后无法得到调色板.pal文件. 一定是我哪里用错了。

`./depat.py PAT0.pat -o pal`

**Reply by palxex** [reply]
是我的疏忽，depat.py忘记生成文档了（但其实可以用-h参数看的）。其设计是输入PAT.mkf，给出调色板号和日/夜（0号和5号调色板，也就是日常和回忆两个，是有日/夜两个状态的，其他没有。默认为日）后直接返回图像格式的调色板（因为pal格式用的太少，偷懒了，另外印象中这种格式转换应该有现成工具）。在usenti里import调色板对话框里选文件类型选择下面的palette from image即可使用导出的图像调色板。
举例：
1. 获取平时白天调色板
`./depat.py -p PAT.MKF -i 0 -o daily.png`
2. 获取回忆夜间调色板
` ./depat.py -p PAT.MKF -i 5 --night -o memory-night.bmp`

**Reply by kaoskey** [reply]
现在遇到的问题是Usenti的，因为Usenti默认背景色是调色板第一个颜色，这就导致黑色像素直接被识别成背景，导出这部分的像素就没了，尴尬。要想同时保持背景透明和黑色像素，我再自己折腾吧，不好意思麻烦您了。再次感谢秒回！

**Reply by palxex** [reply]
哪里，文档确实不够全- -顺便，仙剑用的透明色是每个调色板最后一个颜色（255号），所有工具也是以这个为前提写的。usenti我其实不是太熟，就辛苦你摸索了

**Reply by kaoskey** [reply]
![Palss RGM Test](https://user-images.githubusercontent.com/87409689/126760116-aa5a85b9-00ab-4ad9-9c38-6a52df431cec.png)

用了调色板以后确实好很多了，虽然还有点杂色。再次感谢！

**Reply by palxex** [reply]
很高兴看到有进展:)
另外想问一下，你是直接在usenti里import调色板、打开图片、导出图片、导入仙剑后就直接得到了这个结果吗？还是进行了手工干预。如果是手工干预的结果，可以总结一下你的步骤吗？我加到文档里，省得后来者再走弯路:)

**Reply by kaoskey** [reply]
1. depat.py 导出调色板后使用PalSuite3 (一个调色板修改软件)把第一个黑色换成最后一个颜色，否则黑色像素会翻车 (因为Useinti默认背景色是调色板的第一个颜色)，导致保存的时候损失黑色像素。
2. PalSuite3保存修改的调色板为.pal, 并在Usenti里设置成默认。
3. Usenti 开两个图，一个土星版原图，一个新建使用上面导出的调色版 (先调色板直接导入图片那调整的地方太多了，整个图基本颜色都不对了)。使用Usenti的select功能把土星版原图复制到新建的图上。
4. 使用取色器手动修正一些色差，保存为png (8bit - transparent)
5. 导入仙剑

可能我的方法太过复杂，如果谁知道Usenti怎么换background color就好了，或者有更好用的工具，这样就不用动导出的调色版。否则这样弄完，本来调好的颜色还是会红色溢出，可能就是因为改了调色板的原因。

另外还是期望有没有大神直接从土星原版里重新导出素材（如果直接有土星版的png可能效果更好），我编程小白，这个真无能了，Palss里的东西搞不明白。

再次感谢鲁痴大佬啦！
